### ISSUE TITLE
MatMul flop incorrect for 3D inputs

### ISSUE THREAD

**Issue by ppwwyyxx** [tensorflow/tensorflow#22071]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:N/A
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:N/A
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:v1.10.0-0-g656e7a2b34 1.10.0
- **Python version**:3.7
- **Bazel version (if compiling from source)**:N/A
- **GCC/Compiler version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:N/A
- **GPU model and memory**:N/A
- **Exact command to reproduce**:see below
```python
import tensorflow as tf
a = tf.random_normal([1, 100, 100])
b = tf.random_normal([1, 100, 100])
c = tf.matmul(a, b)
tf.profiler.profile(
            tf.get_default_graph(),
            cmd='op',
            options=tf.profiler.ProfileOptionBuilder.float_operation())
```
The outputs are correct (2M flops) if `a` and `b` have shape `[100, 100]`.
But with 3D inputs, it outputs no flops for matmul.

The reason is that for 3D inputs it uses the "BatchMatMul" op which does not have flop statistics implemented.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
@tatatodd  -  Hi Todd, could you please take a look at this ?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by eyalsh** [comment]
**Tensorflow version:** b'v1.13.1-6-gd32c49d4e3' 1.13.1

The following code which attempts to profile the flops and params of ResNet50 reports 
FLOPS:51,351,635 PARAMS:25,636,712

Params seems ok but flops is way off (should be ~4GFLOPS).
Potentially could be the same issue
 
```python
import tensorflow as tf
import keras.backend as K
from keras.applications.resnet50 import ResNet50

def main():
    run_meta = tf.RunMetadata()
    with tf.Session(graph=tf.Graph()) as sess:
        K.set_session(sess)        
        
        m = ResNet50()

        opts = tf.profiler.ProfileOptionBuilder.float_operation()    
        flops = tf.profiler.profile(sess.graph, run_meta=run_meta, cmd='op', options=opts)

        opts = tf.profiler.ProfileOptionBuilder.trainable_variables_parameter()    
        params = tf.profiler.profile(sess.graph, run_meta=run_meta, cmd='op', options=opts)

        print('FLOPS:{:,} PARAMS:{:,}'.format(flops.total_float_ops, params.total_parameters))

if __name__ == '__main__':
    main()
```

**Comment by csarron** [comment]
Hi @rmlarsen, since this issue hasn't got addressed yet, I created a commit that may solve the BatchMatMul op flops problem, could you help take a look? Thanks.
